activesupport/test/test_case_test.rb

```diff
@@ -236,6 +246,18 @@ def test_assert_changes_with_to_option_but_no_change_has_special_message
     assert_equal "\"@object.num\" didn't change. It was already 0.\nExpected 0 to not be equal to 0.", error.message
   end
 
+  def test_assert_changes_message_with_lambda
+    skip if !defined?(RubyVM)
+
+    error = assert_raises Minitest::Assertion do
+      assert_changes -> { @object.num }, to: 0 do
+        # no changes
+      end
+    end
+
+    assert_equal "\"-> { @object.num }\" didn't change. It was already 0.\nExpected 0 to not be equal to 0.", error.message
```

Comment:
I think I'd prefer something like: 

```suggestion
    assert_equal "`@object.num` didn't change. It was already 0.\nExpected 0 to not be equal to 0.", error.message
```

Reply:
Yeah I like that better as well, it's just a bit harder to implement since we do not get location information about the actual content of the proc. I tried implementing it in the second commit (I'll squash those before merging) using a regex, but I'm not sure how robust this solution is. Let me know what you think!

Reply:
I made a pass on it. Using regexp to parse Ruby code is tricky, so I instead went to a route were we only include the source if we match the happy path (single line, defined with `{}`, no arguments).

Reply:
The idea being that the overwhelming majority of procs used for `assert_changes` match these criteria, and those which don't would be a pain to render properly, so we just fallback to the old behavior.

Reply:
Sounds like a good compromise to me. Thanks for merging!